pages/en/lb3/Attaching-models-to-data-sources.md

```diff
@@ -66,14 +66,66 @@ For example:
 ```javascript
 "corp1": {
     "name": "corp1",
-    "connector": "mysql",
-    "host": "your-mysql-server.foo.com",
+    "connector": "mongodb",
+    "host": "your-mongodb-server.foo.com",
     "user": "db-username",
     "password": "db-password",
     "database": "your-db-name"
   }
 ```
 
+{% include warning.html content="Putting production database credentials in a JSON file is not recommended for security reasons.
+Instead, load the credentials from environment variables, as explained [below](#specifying-database-credentials-with-environment-variables).
+"%}
+
+### Specifying database credentials with environment variables
+
+Best practice is not to put production database credentials explicitly in JSON or JavaScript files, where they could be  a security vulnerability.  Instead, define the values in environment variables and reference them in the data source configuration file.
+
+For example, assuming you have set a valid MongoDB username and password in the
+environment variables MONGO_USER and MONGO_PASS, respectively:
+
+{% include code-caption.html content="datasources.json" %}
+```javascript
+  ...
+  "accountDS": {
+    "name": "accountDS",
+    "connector": "mongodb",
+    "host": "demo.strongloop.com",
+    "port": 27017,
+    "database": "demo",
+    "username": "${MONGO_USER}",
+    "password": "${MONGO_PASS}"
+  }
+  ...
+ ```
+
+You can use different data source definitions (including database credentials) for development and production by using the NODE_ENV
+environment variable, as explained in [Environment-specific configuration](Environment-specific-configuration.html#data-source-configuration),
+for example `datasources.production.json` for production environment when NODE_ENV is 'production'.
+
+### Using multiple data source configurations
+
+LoopBack merges environment-specific configurations (for example in `datasources.production.json`) with the baseline configuration in `datasources.json`.
+Environment-specific configurations can override the top-level values with string and number values only. You cannot use objects or array values.
```

Comment:
This is not true - AFAIK, one can override object values with object values, array values with array values too.